src/pt/mangaterra/src/eu/kanade.tachiyomi.extension.pt.mangaterra/MangaTerra.kt

```diff
@@ -0,0 +1,207 @@
+        setUrlWithoutDomain(element.absUrl("href"))
+    }
+
+    override fun chapterListSelector() = ".card-list-chapter a"
+
+    override fun imageUrlParse(document: Document) = document.selectFirst("img")!!.srcAttr()
+
+    override fun latestUpdatesFromElement(element: Element) = popularMangaFromElement(element)
+
+    override fun latestUpdatesNextPageSelector() = popularMangaNextPageSelector()
+
+    override fun latestUpdatesRequest(page: Int) = GET("$baseUrl/manga?q=u&page=$page", headers)
+
+    override fun latestUpdatesSelector() = popularMangaSelector()
+
+    override fun mangaDetailsParse(document: Document) = SManga.create().apply {
+        title = document.selectFirst(".card-body h1")!!.ownText()
+        description = document.selectFirst(".card-body p")?.ownText()
+        thumbnail_url = document.selectFirst(".card-body img")?.srcAttr()
+        genre = document.select(".card-series-about a").joinToString { it.ownText() }
+    }
+
+    override fun pageListParse(document: Document): List<Page> {
+        val pages = mutableListOf<Page>()
+        var page = 1
+        do {
+            val dom = client.newCall(GET("${document.location()}/${page++}", headers)).execute().asJsoup()
+            pages += Page(index = page, imageUrl = imageUrlParse(dom))
+        } while (dom.selectFirst(".btn-next") != null)
+
+        return pages
+    }
+
+    override fun popularMangaFromElement(element: Element) = SManga.create().apply {
+        title = element.selectFirst("p")!!.ownText()
+        thumbnail_url = element.selectFirst("img")?.srcAttr()
+        setUrlWithoutDomain(element.selectFirst("a")!!.absUrl("href"))
+    }
+
+    override fun popularMangaNextPageSelector() = ".pagination > .page-item:not(.disabled):last-child"
+
+    override fun popularMangaRequest(page: Int) = GET("$baseUrl/manga?q=p&page=$page", headers)
+
+    override fun popularMangaSelector(): String = ".card-body .row > div"
+
+    override fun searchMangaFromElement(element: Element) = popularMangaFromElement(element)
+
+    private fun searchByQueryMangaParse(response: Response): MangasPage {
+        val fragment = Jsoup.parse(
+            json.decodeFromString<String>(response.body.string()),
+            baseUrl,
+            Parser.htmlParser(),
+        )
+
+        return MangasPage(
+            mangas = fragment.select("div.grid-item-series").map(::searchMangaFromElement),
+            hasNextPage = false,
+        )
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage {
+        if (response.request.url.pathSegments.contains("search")) {
+            return searchByQueryMangaParse(response)
+        }
+        return super.searchMangaParse(response)
+    }
+
+    override fun searchMangaNextPageSelector() = popularMangaNextPageSelector()
+
+    override fun fetchSearchManga(page: Int, query: String, filters: FilterList): Observable<MangasPage> {
+        if (query.startsWith(slugPrefix)) {
```

Comment:
Throws `kotlin.UninitializedPropertyAccessException` for `url` when manually searching for `slug:martial-peak`.

Non-existing slug throws `NullPointerException`, but that's probably fine.